finality-aleph/src/new_network/component.rs

```diff
@@ -0,0 +1,42 @@
+use crate::new_network::{DataNetwork, SendError};
+use aleph_bft::Recipient;
+use codec::Codec;
+use futures::channel::mpsc;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+/// For sending arbitrary messages.
+pub trait Sender<D: Clone + Codec + Send>: Clone {
+    fn send(&self, data: D, recipient: Recipient) -> Result<(), SendError>;
+}
+
+/// For receiving arbitrary messages.
+#[async_trait::async_trait]
+pub trait Receiver<D: Clone + Codec + Send>: Sync + Send {
```

Comment:
would it be possible to auto implement `Stream<Item=D>` for the receiver?

Reply:
Probably, but I would prefer not doing that as it would require directly using `Poll`s.